### ISSUE TITLE
.swiftlint.yml is completely ignored

### ISSUE THREAD

**Issue by andrelind** [realm/SwiftLint#229]
I just can't get the .swiftlint.yml to load, I've tried to have it in the root folder of my project, in the built app folder, even in /usr/local/bin but it just won't load...!

I've tried with both a clean build from master branch as well as the 0.4.0 release

swftlint runs as it should but it doesn't ignore the /Pods and /Carthage folders :-(

My file
`excluded: # paths to ignore during linting. overridden by `included`.`
`	- Carthage`
`	- Pods`
`# parameterized rules can be customized from this configuration file`
`line_length: 200`
`# parameterized rules are first parameterized as a warning level, then error level.`
`type_body_length:`
`	- 300 # warning`
`	- 400 # error`
`# parameterized rules are first parameterized as a warning level, then error level.`
`variable_name_max_length:`
`	- 40 # warning`
`	- 60 # error`
`# parameterized rules are first parameterized as a warning level, then error level.`
`variable_name_min_length:`
`	- 3 # warning`
`	- 2 # error`

**Comment by bensarz** [comment]
Having the same problem.
Is there a `brew` way of returning to 0.3.0?

**Comment by jpsim** [comment]
Are you running `swiftlint` from the same directory in which `.swiftlint.yml` is located?

When you run `swiftlint`, does it print something like "Loading configuration from 'config path'" to stderr?

If not, then it means the configuration file cannot be found. You can specify a path to your configuration file using the `--config [path to .swiftlint.yml]` command line flag.

**Comment by bensarz** [comment]
I have it setup properly as per the README instructions.
My file is named `.swiftlint.yml` and it is in the root directory of the project, side by side with the `.xcworkspace`.

I tried passing the path to `swiftlint` like this:

```
$ cd MyProject
$ swiftlint --config .swiftlint.yml
Unrecognized command: '--config'. See `swiftlint help`.
```

As you can see, I'm getting an error regarding unrecognized command.

**Comment by bensarz** [comment]
Ok, I'm obviously speaking before thinking here.

I ran this command successfully:

```
$ swiftlint lint --config .swiftlint.yml
```

There's the content of my `.swiftlint.yml` file:

```yaml
disabled_rules: # rule identifiers to exclude from running
#  - colon
#  - comma
#  - control_statement
#  - file_length
#  - force_cast
#  - force_try
#  - function_body_length
#  - leading_whitespace
  - line_length
  - nesting
#  - opening_brace
#  - operator_whitespace
#  - return_arrow_whitespace
#  - statement_position
#  - todo
#  - trailing_newline
#  - trailing_semicolon
   - trailing_whitespace
#  - type_body_length
#  - type_name
#  - variable_name_max_length
#  - variable_name_min_length
#  - variable_name
included: # paths to include during linting. `--path` is ignored if present. takes precendence over `excluded`.
  - MyProject
  - MyProjectKeyboard
  - MyProjectTests
excluded: # paths to ignore during linting. overridden by `included`.
  - Carthage
  - Pods

file_length:
  - 1000
function_body_length:
  - 75
  - 100
type_body_length:
  - 400 # Warning
  - 500 # Error
```

I still get a bunch of these:

```
../Pods/Alamofire/Source/Alamofire.swift:28: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
../Pods/Alamofire/Source/Alamofire.swift:356: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
../Pods/Alamofire/Source/Alamofire.swift:360: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
```

The `Pods` folder should be ignore as well as the `Trailing Whitespace` rule.

**Comment by jpsim** [comment]
Looks like SwiftLint isn't installed properly on your system (though I make no assumptions as to how it got this way). After a fresh installation, here's what swiftlint outputs for me:

```shell
$ which swiftlint
$ brew install swiftlint
==> Downloading https://homebrew.bintray.com/bottles/swiftlint-0.4.0.el_capitan.bottle.tar.gz
Already downloaded: /Library/Caches/Homebrew/swiftlint-0.4.0.el_capitan.bottle.tar.gz
==> Pouring swiftlint-0.4.0.el_capitan.bottle.tar.gz
🍺  /usr/local/Cellar/swiftlint/0.4.0: 215 files, 55M
$ swiftlint version
0.4.0
$ swiftlint help lint
Print lint warnings and errors for the Swift files in the current directory (default command)

[--path (string)]
	the path to the file or directory to lint

[--use-stdin]
	lint standard input

[--config .swiftlint.yml]
	the path to SwiftLint's configuration file

[--strict]
	fail on warnings

[--use-script-input-files]
	read SCRIPT_INPUT_FILE* environment variables as files
```

See how `--config` is a valid option here?

If you ever installed SwiftLint *from source*, you may have mismatched versions of the framework & binary. So please uninstall it (`make uninstall`). Also uninstall it from homebrew `brew uninstall swiftlint`.

Then make sure your Homebrew repository is up to date: `brew update`. Then re-install SwiftLint via Homebrew: `brew install swiftlint`.

Let me know how that goes.

**Comment by bensarz** [comment]
I reinstall `SwiftLint`. These are the commands I ran (copy/pasted from my Terminal):

```bash
$ which swiftlint
/usr/local/bin/swiftlint
$ swiftlint version
0.4.0
$ brew uninstall swiftlint
Uninstalling /usr/local/Cellar/swiftlint/0.4.0... (215 files, 55M)
$ which swiftlint
swiftlint not found
$ brew update
Already up-to-date.
$ brew install swiftlint
==> Downloading https://homebrew.bintray.com/bottles/swiftlint-0.4.0.yosemite.bottle.tar.gz
Already downloaded: /Library/Caches/Homebrew/swiftlint-0.4.0.yosemite.bottle.tar.gz
==> Pouring swiftlint-0.4.0.yosemite.bottle.tar.gz
🍺  /usr/local/Cellar/swiftlint/0.4.0: 215 files, 55M
$ which swiftlint
/usr/local/bin/swiftlint
$ swiftlint version
0.4.0
```

When testing this with a new project, everything seems to be working fine.
Look like the problem is tied to my existing project.
Will dig deeper and report my findings.

**Comment by jpsim** [comment]
Thanks for working this through with me :smile:. The fact that Swift frameworks can't be statically linked is the cause of issues like these, leading frameworks & the apps that link them to some times get out of sync.

**Comment by bensarz** [comment]
No problem, you're always ready to help with Realm and SwiftLint stuff. It's only fair to give back to you!
So here's what I have found.

I am using `Cocoapods` with and making use of `use_framework!` (not sure if this applies, but it has been a source of other problems, so I'm mentioning it here). I started by deleting my `.xcworkspace` file, deleting the `Pods/` folder, deleted the `Podfile.lock` to ensure the latest versions of dependencies. The content of my `.swiftlint.yml` hasn't changed. Built the project and it started working.

Commenting out `- trailing_whitespace` back and forth has the desired effect so I know it's working.
I wish I could provide more than "it just started working" but it definitely seems related to the workspace file somehow.

**Comment by jpsim** [comment]
> I wish I could provide more than "it just started working" but it definitely seems related to the workspace file somehow.

I have absolutely no idea why the presence of a `.xcworkspace` would affect SwiftLint's behavior in that way. In fact, we use a workspace for the development of SwiftLint itself with no issues.

**Comment by bensarz** [comment]
I'm only guessing but could it be related to `xcuserdata/bensarz.xcuserdatad/UserInterfaceState.xcuserstate` or `contents.xcworkspacedata` inside the workspace?

**Comment by jpsim** [comment]
That seems entirely impossible, but hey, I don't know what's happening at all here so ¯\\\_(ツ)\_/¯.

**Comment by bensarz** [comment]
I take it back. Tried to move onto my coworker's computer and it failed again. :dizzy_face:
Will keep digging.

**Comment by jpsim** [comment]
FYI, you can run SwiftLint from Xcode, so you can set breakpoints, use LLDB, etc. Just edit the `swiftlint`'s build scheme's "Options -> Working Directory" and its "Arguments -> Arguments Passed on Launch".

**Comment by bensarz** [comment]
I was able to fix it by properly formatting our YAML file.
![image](https://cloud.githubusercontent.com/assets/1077441/11347316/e0492314-91ee-11e5-915f-c34a990fb9bf.png)

Notice how the `- trailing_whispace` has 3 spaces instead of 2.
Everything seems to be working fine now.

Maybe this would be a good enhancement: validate the YAML file if found?

**Comment by jpsim** [comment]
I'm happy your figured this out!

> Maybe this would be a good enhancement: validate the YAML file if found?

Yes! We're tracking this as #120, and I'd appreciate some help doing this, if you're interested in helping us out.

**Comment by bensarz** [comment]
Ok awesome!
Thank you, I will try to dedicate some time to this.

**Comment by jpsim** [comment]
@andrelind is it possible your YAML file is also formatted incorrectly? You can validate your file here: http://www.yamllint.com/

**Comment by andrelind** [comment]
> Notice how the - trailing_whispace has 3 spaces instead of 2.
> Everything seems to be working fine now.

This...! Now it loads the config file as it should! :+1: 
Apparently copying the text from the realm file had tabs instead of spaces!

**Comment by jpsim** [comment]
I'm glad you both figured it out.

**Comment by drrost** [comment]
In my case, I had to delete spaces before the rules dash, so it looks like this:

```
disabled_rules:
- identifier_name
```

Swiftlint 0.34.0 is used.
